components/rowListScene.brs

```diff
@@ -1,12 +1,23 @@
 sub init()
-    m.global.addField("http", "node", FALSE)
     m.global.addField("ratio", "float", FALSE)
-    m.global.http = createObject("roSGNode", "httpTask")
-    m.global.http.control = "RUN"
+    m.flag = true
     m.global.ratio = 1
     m.rowList = m.top.findNode("mainRowList")
+    m.heroDetailsId = m.top.findNode("heroDetailsId")
+    m.MainBoldLabel = m.heroDetailsId.findNode("MainBoldLabel")
+    m.DateAndDurationLabel = m.heroDetailsId.findNode("DateAndDurationLabel")
+    m.DescriptionLabel = m.heroDetailsId.findNode("DescriptionLabel")
+    m.CategoryLabel = m.heroDetailsId.findNode("CategoryLabel")
+    m.fadeInAnimation = m.top.findNode("fadeInAnimation")
+    m.fadeOutAnimation = m.top.findNode("fadeOutAnimation")
     m.rowList.content = createObject("roSGNode", "RowListContent")
     m.rowList.setFocus(true)
+    m.top.observeField("HeroDetails", "handleHeroDetails")
+    m.rowList.observeField("rowItemSelected", "handleItemSelected")
+    m.fadeOutAnimation.observeField("state", "handleAnimationDone")
+
+
+    ' Handling Responsivness
     videoMode = createObject("roDeviceInfo")
     if videoMode.GetVideoMode() = "720p"
```

Comment:
I will simplify this code for you when I have time. There is a better way to do it.